src/frontend/src/app/problem/problem.component.ts

```diff
@@ -1,37 +1,89 @@
-import { Component, OnInit } from '@angular/core';
-import { ActivatedRoute, Router } from '@angular/router';
+import { Component, Inject, OnDestroy, OnInit } from '@angular/core';
+import { ActivatedRoute, ParamMap, Router } from '@angular/router';
+import { LobbyService } from '../services/lobby/lobby.service';
 import { ProblemsService } from '../services/problems/problems.service';
+import { EndData } from '../types/EndData';
+import { PlayerData } from '../types/PlayerData';
+import {
+  MatDialog,
+  MAT_DIALOG_DATA,
+  MatDialogRef,
+} from '@angular/material/dialog';
+import { Subscription } from 'rxjs';
 
 @Component({
   selector: 'app-problem',
   templateUrl: './problem.component.html',
   styleUrls: ['./problem.component.scss'],
 })
-export class ProblemComponent implements OnInit {
+export class ProblemComponent implements OnInit, OnDestroy {
   title: string = '';
   description: string = '';
   id: string = '';
 
+  players: PlayerData[] = [];
+  endDataSub: Subscription | undefined;
+
   constructor(
     private problemService: ProblemsService,
     private aRoute: ActivatedRoute,
-    private router: Router
+    private router: Router,
+    private lobbyService: LobbyService,
+    public dialog: MatDialog
   ) {}
 
   ngOnInit() {
-    const id = this.aRoute.snapshot.paramMap.get('id');
-    if (id != null) {
-      this.problemService.getProblems().subscribe((problems) => {
-        for (var i = 0; i < problems.length; i++) {
-          if (problems[i]._id === id) {
-            this.title = problems[i].name;
-            this.description = problems[i].description;
-            this.id = problems[i]._id;
+    this.lobbyService.waitingRoom.subscribe((players) => {
```

Comment:
Extract each subscription to a local function

Reply:
we can fix this in #90 since that code was there before this pr